src/shims/env.rs

```diff
@@ -110,8 +110,24 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         }
     }
 
+    /// Get the process identifier.
     fn get_pid(&self) -> u32 {
         let this = self.eval_context_ref();
         if this.machine.communicate() { std::process::id() } else { 1000 }
     }
+
+    /// Get an "OS" thread ID for the current thread.
+    fn get_current_tid(&self) -> u32 {
+        self.get_tid(self.eval_context_ref().machine.threads.active_thread())
+    }
+
+    /// Get an "OS" thread ID for any thread.
+    fn get_tid(&self, thread: ThreadId) -> u32 {
+        let this = self.eval_context_ref();
+        let index = thread.to_u32();
+
+        // todo: is this also true on Windows?
+        // Compute a TID for this thread, ensuring that the main thread has PID == TID.
+        this.get_pid().strict_add(index)
+    }
```

Comment:
@ChrisDenton is there any relationship between the PID and TIDs on Windows?

This (preexisting) logic works for Unix because the first TID for a process is the same as the PID, but if that logic isn't true on Windows then I'll need to update this. If it is true on Windows, then I think the `CreateThread` shim needs to get updated to return the same TID logic in `lpThreadId` (hence the failing test).

Reply:
On Windows thread IDs and process IDs are unrelated.

Reply:
But it doesn't hurt to use the Unix scheme on Windows as well I guess?

Reply:
Oops, I replied before seeing your message Ralf. Apparently it seems like a Linux thing more specifically than a Unix thing, I can't find documentation for other platforms about a relationship between PID and TID.

I just updated to only offset by PID on Linux, but do you prefer I put it back on all platforms and update the address saved by `CreateThread` instead?

Reply:
Just checked a handful of different platforms on cfarm:

Linux

```text
main tid 2212321 0x21c1e1
main pid 2212321 0x21c1e1
```

FreeBSD: 

```text
main tid 0 0
main pid 23750 0x5cc6
```

OpenBSD:

```text
main tid 137368 0x21898
main pid 93069 0x16b8d
```

NetBSD:

```text
main tid 5057 0x13c1
main pid 5057 0x13c1
```

MacOS:

```text
main tid 7763176 0x7674e8
main pid 51507 0xc933
```

Based on that nonscientific experiment I'll add NetBSD to the list of platforms that says tid is an offset from pid. For the others it's probably actually better that they don't line up, so users don't rely on that non-cross-platform behavior.